openwisp_monitoring/monitoring/models.py

```diff
@@ -239,6 +239,16 @@ def _set_extra_notification_opts(self, opts):
                                  args=[target.pk])
             opts['data']['url'] = '{}{}'.format(base_url, device_url)
 
+    @classmethod
+    def delete_related_metric(cls, instance, **kwargs):
+        """
+        Deletes metrics related to a device upon device's deletion
+        """
+        device = instance
+        metrics = cls.objects.filter(object_id=device.pk)
+        if metrics:
+            metrics.delete()
```

Comment:
same as in checks, simplify the delete operation (can be done in 1 line) and move it to apps.py to make the whole flow easier to read